core/src/main/java/io/aiven/klaw/service/TopicControllerService.java

```diff
@@ -650,6 +675,25 @@ private List<TopicRequestModel> getTopicRequestModels(
     return topicRequestModelList;
   }
 
+  private void validateAndCopyTopicConfigs(
+      TopicRequest topicReq, TopicRequestModel topicRequestModel) {
+    try {
+      List<String> advancedTopicConfigs = new ArrayList<>();
+      if (topicReq.getJsonParams() != null) {
+        TopicConfigurationRequest topicConfigurationRequest =
+            OBJECT_MAPPER.readValue(topicReq.getJsonParams(), TopicConfigurationRequest.class);
+        for (Map.Entry<String, String> entry :
+            topicConfigurationRequest.getAdvancedTopicConfiguration().entrySet()) {
+          advancedTopicConfigs.add(entry.getKey() + ":" + entry.getValue());
```

Comment:
Do we really need this manipulations with strings and `:`?
Could we just use POJO?

Reply:
It is a pojo now. Earlier there were issues from frontend sending data.